os/kernel/binary_manager/binary_manager_getinfo.c

```diff
@@ -135,6 +135,41 @@ int binary_manager_get_available_size(int bin_idx)
 	return size;
 }
 
+/****************************************************************************
+ * Name: binary_manager_get_state
+ *
+ * Description:
+ *	 This function gets binary state with binary name.
+ *
+ ****************************************************************************/
+void binary_manager_get_state_with_name(int requester_pid, char *bin_name)
+{
+	int bin_idx;
+	uint32_t bin_count;
+	char q_name[BIN_PRIVMQ_LEN];
+	binmgr_getstate_response_t response_msg;
+
+	if (requester_pid < 0 || bin_name == NULL) {
+		bmdbg("Invalid data pid %d name %s\n", requester_pid, bin_name);
+		return;
+	}
+	snprintf(q_name, BIN_PRIVMQ_LEN, "%s%d", BINMGR_RESPONSE_MQ_PREFIX, requester_pid);
+
+	memset((void *)&response_msg, 0, sizeof(binmgr_getstate_response_t));
+	response_msg.result = BINMGR_NOT_FOUND;
+
+	bin_count = binary_manager_get_ucount();
+	for (bin_idx = 1; bin_idx <= bin_count; bin_idx++) {
+		if (!strncmp(BIN_NAME(bin_idx), bin_name, BIN_NAME_MAX)) {
```

Comment:
I think you'd better compare with `strlen(bin_name) + 1` instead of `BIN_NAME_MAX`.